include/graphit/frontend/fir.h

```diff
@@ -1654,6 +1656,22 @@ namespace graphit {
                virtual void copy(FIRNode::Ptr);
         };
 
+        struct FuncExpr : public Expr {
```

Comment:
Do we need another FIR node for the FuncExpr parameter? Since all these nodes are expressions, can we not use Call and make the name/target as an expression?

Reply:
I think we certainly can do that. But isn't it more clear if the functor belongs to a different FIR node than call expr? Especially if i decide to push it all the way down to backend

Reply:
This is a bit different from CallExpr. This is more about passing a function as an argument and not calling the function.